fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/v2/tools/reactive/RxTools.kt

```diff
@@ -22,28 +21,14 @@ fun completable(function: () -> Unit): Completable = Completable.fromAction {
     function.invoke()
 }
 
-inline fun <reified R> Flowable<*>.filterCast(
-    crossinline predicate: (R) -> Boolean = { true }
-) =
-    this.filter { it is R && predicate(it) }
-        .map { it as R }
+inline fun <reified R> Flowable<*>.filterCast() = this.filter { it is R }.map { it as R }
```

Comment:
I think you could also do it with a single operation `.mapNotNull { it as? R }`

Reply:
mapNotNull is not part of RX apis yet